### PR TITLE
docs: document file resource endpoints

### PR DESCRIPTION
**Pull request by carlosmunoz**
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-api/53)

### CODE REVIEW

File: zanata-common-api/src/main/java/org/zanata/rest/service/FileResource.java [hunk 1]
```diff
@@ -92,43 +92,75 @@
...
+     *
+     * @param projectSlug The project slug where to store the document.
+     * @param iterationSlug The project version slug where to store the document.
+     * @param localeId The locale (language) for the translation file.
+     * @param docId The full Document identifier.
+     * @param merge Indicates whether to merge translations or overwrite all
+     *              translations with the contents of the uploaded file.
+     * @param uploadForm The multi-part form body for the file or chunk.
+     * @return A message with information about the upload operation.
+     * @return
```
Comment (djansen-redhat) [inline zanata-common-api/src/main/java/org/zanata/rest/service/FileResource.java]: Two @returns

**Comment by carlosmunoz** [comment]
@djansen-redhat thanks for bringing this back to my attention
@davidmason can you please review this?

**Comment by davidmason** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[zanata-common-api/src/main/java/org/zanata/rest/service/FileResource.java, line 138 \[r2\]](https://reviewable.io:443/reviews/zanata/zanata-api/53#-KO2oQ8GqZp-Mnb1u3rR:-KO2oQ8GqZp-Mnb1u3rS:-395527488) ([raw file](https://github.com/zanata/zanata-api/blob/69532400ac236659b2688c9fed578990f75058b9/zanata-common-api/src/main/java/org/zanata/rest/service/FileResource.java#L138)):*
> ```Java
>      * into smaller chunks for very large files. In this case, the first invocation
>      * of this service will return an 'upload id' which needs to be used in
>      * subsequent calls to tie all the uploaded chunks together.
> ```

For this and the above one, we could add a note that the file will not be processed until all chunks are uploaded. There is a field in DocumentFileUploadForm that indicates when the last chunk is sent, so maybe link to that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-api/53)*

**Comment by davidmason** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-api/53)*

**Comment by carlosmunoz** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[zanata-common-api/src/main/java/org/zanata/rest/service/FileResource.java, line 148 \[r1\]](https://reviewable.io:443/reviews/zanata/zanata-api/53#-KIR0A6HmeH3WQUKffB6-r1-148:-KO7leVzcJHkKCw10JK6:-499146009) ([raw file](https://github.com/zanata/zanata-api/blob/a4d56ee519a41ac42c8eb59dc7ac12c494bda8d8/zanata-common-api/src/main/java/org/zanata/rest/service/FileResource.java#L148)):*
<details><summary><i>Previously, djansen-redhat (Damian Jansen) wrote…</i></summary>
> Two ```@return``` lines
</details>
Done.

---

*[zanata-common-api/src/main/java/org/zanata/rest/service/FileResource.java, line 138 \[r2\]](https://reviewable.io:443/reviews/zanata/zanata-api/53#-KO2oQ8GqZp-Mnb1u3rR:-KO7nSiyi4Wi37ZrD3pi:-499146009) ([raw file](https://github.com/zanata/zanata-api/blob/69532400ac236659b2688c9fed578990f75058b9/zanata-common-api/src/main/java/org/zanata/rest/service/FileResource.java#L138)):*
<details><summary><i>Previously, davidmason (David Mason) wrote…</i></summary>
> For this and the above one, we could add a note that the file will not be processed until all chunks are uploaded. There is a field in DocumentFileUploadForm that indicates when the last chunk is sent, so maybe link to that.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-api/53)*

**Comment by davidmason** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-api/53#-:-KO7o8N-nj7Nl3a3zSne:1417088721)*

**Pull request merged**